core/src/main/java/com/radixdlt/consensus/liveness/Pacemaker.java

```diff
@@ -280,82 +219,165 @@ public void processBFTUpdate(BFTInsertUpdate update) {
+      // If the round can be prolonged, then do so.
+      // Do not send any timeout votes and/or create fallback vertex yet.
+      // Note that even though the round has been prolonged (i.e. a "real" round timeout delayed)
+      // some behaviour has already been altered by setting the `scheduledRoundTimeoutHasOccurred`
+      // flag to true above.
+      // See: `processBFTUpdate` and `dispatchVote`
       prolongRoundTimeout(scheduledTimeout);
     } else {
+      // The round can't be prolonged, so timeout for real
       this.roundStatus = RoundStatus.TIMED_OUT;
       updateTimeoutCounters(scheduledTimeout);
-      resendPreviousOrNewVoteWithTimeout(this.validatorSet.nodes());
+      /* Re-send a previous vote (if there is one) or
+      insert a fallback vertex and await an async BFT insert update event.
+      See: `processBFTUpdate` */
+      resendPreviousVoteWithTimeoutOrVoteForFallbackVertex();
       // Dispatch an actual timeout occurrence
       this.timeoutDispatcher.dispatch(new LocalTimeoutOccurrence(scheduledTimeout));
       rescheduleTimeout(scheduledTimeout);
     }
   }
 
-  private void prolongRoundTimeout(ScheduledLocalTimeout originalScheduledLocalTimeout) {
-    metrics.bft().prolongedRoundTimeouts().inc();
-    final var timeout = timeoutCalculator.additionalRoundTimeIfProposalReceivedMs();
-    final var nextTimeout = originalScheduledLocalTimeout.prolong(timeout);
-    this.scheduledLocalTimeoutDispatcher.dispatch(nextTimeout, timeout);
+  /**
+   * Attempts to re-send a previously sent vote (if present) with a timeout flag included. If no
+   * vote has been sent yet, attempts to send a vote for a fallback vertex (if present). If the
+   * fallback vertex doesn't exist yet, then attempts to create one and returns. The insertion of a
+   * fallback vertex should trigger an async BFT update event and the "vote for a fallback vertex"
+   * process should carry on from there (see: `processBFTUpdate`).
+   */
+  private void resendPreviousVoteWithTimeoutOrVoteForFallbackVertex() {
+    this.safetyRules
+        .getLastVote(currentRound())
+        .map(this.safetyRules::timeoutVote)
+        .ifPresentOrElse(
+            this::dispatchVote,
+            () -> voteForFallbackVertexOrInsertIfMissing(this.latestRoundUpdate));
+  }
+
+  /**
+   * A helper for `resendPreviousVoteWithTimeoutOrVoteForFallbackVertex`, see its doc for details
+   */
+  private void voteForFallbackVertexOrInsertIfMissing(RoundUpdate roundUpdate) {
+    final var highQC = this.latestRoundUpdate.getHighQC();
+    final var vertex =
+        Vertex.createFallback(
+                highQC.highestQC(), roundUpdate.getCurrentRound(), roundUpdate.getLeader())
+            .withId(hasher);
+
+    this.vertexStore
+        .getExecutedVertex(vertex.hash())
+        .ifPresentOrElse(
+            this::attemptVoteOnVertex,
+            () -> {
+              // Fallback vertex doesn't yet exist, so try inserting one
```

Comment:
Perhaps it's worth briefly mentioning here that we'll vote on it when we received the BFTInsertUpdate